core/src/apps/management/reset_device/layout.py

```diff
@@ -355,13 +400,33 @@ def _split_share_into_pages(share_words):
     return count
 
 
-async def slip39_prompt_number_of_groups(ctx):
+async def slip39_basic_show_and_confirm_shares(ctx, shares):
+    # warn user about mnemonic safety
+    await show_backup_warning(ctx, slip39=True)
+
+    for index, share in enumerate(shares):
+        share_words = share.split(" ")
+        while True:
+            # display paginated share on the screen
+            await _show_share_words(ctx, share_words, index)
+
+            # make the user confirm words from the share
+            if await _confirm_share_words(ctx, index, share_words):
+                await _show_confirmation_success(
+                    ctx, index, num_of_shares=len(shares), slip39=True
+                )
+                break  # this share is confirmed, go to next one
+            else:
+                await _show_confirmation_failure(ctx, index)
```

Comment:
This internal while loop is very similar to the one in `bip39_show_and_confirm_mnemonic`. Is it worth it to merge these? Couldn't we have a single `show_and_confirm_shares` for both actually?

Reply:
I was thinking about it but we can only unify bip39 and slip39_basic, slip39_advanced has one extra loop that would be tricky to unify. So we still end up with two functions. The actual display of the words and their confirmations is unified which I think was the main goal of #409 .

Reply:
Fair enough.